core/src/main/scala/commandcenter/command/CalculatorCommand.scala

```diff
@@ -1,29 +1,129 @@
+      evaluatedValue <- ZIO
+                          .fromOption(new CalculatorUtil(parameters).evaluate(searchInput.input))
+                          .orElseFail(CommandError.NotApplicable)
     } yield List(
       Preview(evaluatedValue)
-        .onRun(tools.setClipboard(evaluatedValue.toString))
+        .render(parameters.format)
+        .onRun(tools.setClipboard(parameters.format(evaluatedValue)))
         .score(Scores.high(searchInput.context))
     )
 }
 
 object CalculatorCommand extends CommandPlugin[CalculatorCommand] {
-  def make(config: Config): TaskManaged[CalculatorCommand] = ZManaged.succeed(CalculatorCommand())
+  def make(config: Config): TaskManaged[CalculatorCommand] = ZManaged.fromEither(config.as[CalculatorCommand])
+
+  implicit val decoder: Decoder[CalculatorCommand] = Decoder.instance { c =>
+    val decimalFormat        = new DecimalFormat()
+    val decimalFormatSymbols = decimalFormat.getDecimalFormatSymbols
+    for {
+      decimalSeparator      <- c.get[Option[Char]]("decimalSeparator")
+      groupingSeparator     <- c.get[Option[Char]]("groupingSeparator")
+      parameterSeparator    <- c.get[Option[Char]]("parameterSeparator")
+      groupingSize          <- c.get[Option[Int]]("groupingSize")
+      groupingUsed          <- c.get[Option[Boolean]]("groupingUsed")
+      maximumFractionDigits <- c.get[Option[Int]]("maximumFractionDigits")
+    } yield CalculatorCommand(
+      Parameters(
+        decimalSeparator.getOrElse(decimalFormatSymbols.getDecimalSeparator),
+        groupingSeparator.getOrElse(decimalFormatSymbols.getGroupingSeparator),
+        parameterSeparator.getOrElse(';'),
+        groupingSize.getOrElse(decimalFormat.getGroupingSize),
+        groupingUsed.getOrElse(decimalFormat.isGroupingUsed),
+        maximumFractionDigits.getOrElse(decimalFormat.getMaximumFractionDigits)
+      )
+    )
+  }
+
+  def helpMessageByType(helpType: HelpType): fansi.Str =
+    helpType match {
+      case FunctionsList  => CalculatorUtil.helpMessageFunctionsList
+      case ParametersList => CalculatorUtil.helpMessageParametersList
+    }
+}
+
+final case class Parameters(
+  decimalSeparator: Char,
+  groupingSeparator: Char,
+  parameterSeparator: Char,
+  groupingSize: Int,
+  groupingUsed: Boolean,
+  maximumFractionDigits: Int
+) {
+  def decimalFormat: DecimalFormat =
+    new DecimalFormat() {
+      setDecimalFormatSymbols(new DecimalFormatSymbols(Locale.getDefault) {
+        setDecimalSeparator(decimalSeparator)
+        setGroupingSeparator(groupingSeparator)
+      })
+      setGroupingSize(groupingSize)
+      setGroupingUsed(groupingUsed)
+      setMaximumFractionDigits(maximumFractionDigits)
+      setParseBigDecimal(true)
+    }
+
+  def format(value: BigDecimal): String = {
+    val format = decimalFormat
+    if (value.isWhole) format.setMaximumFractionDigits(0)
+    format.format(value)
```

Comment:
I think `DecimalFormat` only knows how to format Java's BigDecimal, not Scala's. You might have to change this to `format.format(value.bigDecimal)` instead. Otherwise a large `maximumFractionDigits` won't work.

There might be some alternate solution than what I proposed, but it's the only thing I could think of.

Reply:
Hmm.. actually formatting works well as far as I can tell from my testing :). But I see your point and will change it.

Reply:
Hmm. I wonder why. Do you see what I'm seeing with this?

```scala
val df = new DecimalFormat()
df.setMinimumFractionDigits(25)
println(df.format(BigDecimal(1) / BigDecimal(3)))
println(df.format((BigDecimal(1) / BigDecimal(3)).bigDecimal))
```

I get:

```
0.3333333333333333000000000
0.3333333333333333333333333
```

Reply:
Sorry, I meant to use `setMaximumFractionDigits`. But same basic problem:

```
0.3333333333333333
0.3333333333333333333333333
```

Reply:
Oh sorry, my bad. You're absolutely right. Seems I have been more concentrated on the other parameters. Thanks.